lib/internal/test_runner/runner.js

```diff
@@ -248,14 +214,65 @@ class FileTest extends Test {
       this.#buffer = [];
     }
   }
-  addToReport(ast) {
-    this.#accumulateReportItem(ast);
+  addToReport(item) {
+    this.#accumulateReportItem(item);
     if (!this.isClearToSend()) {
-      ArrayPrototypePush(this.#buffer, ast);
+      ArrayPrototypePush(this.#buffer, item);
       return;
     }
     this.#drainBuffer();
-    this.#handleReportItem(ast);
+    this.#handleReportItem(item);
+  }
+  parseMessage(readData) {
+    if (readData.length === 0) return;
+
+    ArrayPrototypePush(this.#messageBuffer, readData);
+    this.#messageBufferSize += readData.length;
+
+    // Index 0 should always be present because we just pushed data into it.
+    let messageBufferHead = this.#messageBuffer[0];
+
+    while (messageBufferHead.length >= 4) {
```

Comment:
this code is based (and partially copied) from `child_process` ipc advanced mode